### DISCUSSION TITLE
REMAP上でキーマップを認識しない（左仕様）。

### DISCUSSION THREAD

**Discussion by yuruna1080** [Yowkees/keyball#74] [Q&A]
お世話になります。
掲題の件でご質問がございましてご連絡いたしました。

keyball46をトラックボール左仕様で作成し、
下記の現時点で最新のファームウェアと思われるVIA対応「keyball_via_Left_1.1.0.zip」をqmk_toolboxで取り込みましたが、
REMAP上では接続の認識がされるものの、JSONファイルを要求されます。

左仕様の場合には別途JSONファイルを作成の上、REMAP上での取り込みが必要と理解で大丈夫でしょうか？
（初めて自作するもので可能であれば右仕様のファームウェアのようにファームウェア取り込みだけでマッピングができればと考えております。。）

なお、補足といたしまして右仕様用のファームウェアを取り込んでみたところ、
REMAP上では接続およびキーレイアウトが認識されてされており、キーマッピングが可能となっておりました。
（ファームウェアversion1.0.0でも上記と同様でした。）
キーやトラックボールはすべて反応することを確認しておりますのでキーボード上に問題はないかと思っています。

初心者で説明不足の部分がありっご迷惑をおかけしますっが、
お手すきの際にご確認いただけますと幸いです。

以下使用サイト等
◇REMAP
https://remap-keys.app/
◇取り込んだファームウェア
https://github.com/Yowkees/keyball/discussions/70
◇qmk_toolbox0.1.1
https://github.com/qmk/qmk_toolbox/releases

**Comment by koron** [comment]
> 左仕様の場合には別途JSONファイルを作成の上、REMAP上での取り込みが必要と理解で大丈夫でしょうか？

左ボール用のJSONファイルは以下のURLにありますのでダウンロードして使ってみてください。

https://raw.githubusercontent.com/Yowkees/keyball/8406200bbf993825890901ab66c04b78dd37acef/qmk_firmware/keyboards/keyball/rev1/keyball46_left.json

YowkeesさんのほうにてRemap側に上記JSONを取り込んでもらっていれば、自動で判定されると期待できます。
もしもそうなっていないのだとすれば何かしらの問題があるのだと推測できますが、
特定のハードウェアや組み立て方法によるものではなくRemap上で起こっていることだと考えてよいかと。

いずれにせよ上記JSONで対応できると思われますのでお試しください。

なおファームウェアについて、ボールがある側にはball版を、無い側にはnoball版を書くようにしてください。
